docs/source-app/workflows/mount_cloud_object_store.rst

```diff
@@ -0,0 +1,150 @@
+:orphan:
+
+################
+Mount Cloud Data
+################
+
+**Audience:** Users who want to read files stored in a Cloud Object Bucket in an app.
+
+******************************
+Mounting Public AWS S3 Buckets
+******************************
+
+===================
+Add Mount to a Work
+===================
+
+To mount data from a cloud bucket to your app compute, initialize a :class:`~lightning_app.storage.mount.Mount`
+object with the source path of the s3 bucket and the absolute directory path where it should be mounted and
+pass the :class:`~lightning_app.storage.mount.Mount` to the :class:`~lightning_app.utilities.packaging.cloud_compute.CloudCompute`
+of the :class:`~lightning_app.core.work.LightningWork` it should be mounted on.
+
+In this example, we will mount an S3 bucket: ``s3://ryft-public-sample-data/esRedditJson/`` to ``/content/esRedditJson/``.
+
+.. code:: python
+    :emphasize-lines: 9-12
+
+    import lightning as L
+    from lightning_app import CloudCompute
+    from lightning_app.storage import Mount
+
+    class Flow(L.LightningFlow):
+       def __init__(self):
+           super().__init__()
+           self.my_work = MyWorkClass(
+               cloud_compute=CloudCompute(
+                   mounts=Mount(
+                       source="s3://ryft-public-sample-data/esRedditJson/",
+                       mount_path="/content/esRedditJson/",
+                   ),
+               )
+           )
+
+       def run(self):
+           self.my_work.run()
+
+You can also pass multiple mounts to a single work by passing a ``List[Mount(...), ...]`` to the
+``CloudCompute(mounts=...)`` argument.
+
+.. note::
+
+    * Mounts are limited to 1M files.
+    * Mount objects should not be larger than 5 GiB.
+    * When adding multiple mounts, each one should have a unique ``mount_path``.
+    * A maximum of 10 :class:`~lightning_app.storage.mount.Mount`\s can be added to a :class:`~lightning_app.core.work.LightningWork`.
+
+=======================
+Read Files From a Mount
+=======================
+
+Once a :class:`~lightning_app.storage.mount.Mount` object is passed to :class:`~lightning_app.utilities.packaging.cloud_compute.CloudCompute`,
+you can access, list, or read any file from the mount under the specified ``mount_path``, just like you would if it
+was on your local machine.
+
+Assuming your ``mount_path`` is ``"/content/esRedditJson/"`` you can do the following:
+
+----------
+Read Files
+----------
+
+.. code:: python
+
+    with open("/content/esRedditJson/esRedditJson1", "r") as f:
+        some_data = f.read()
+
+    # do something with "some_data"...
+
+----------
+List Files
+----------
+
+.. code:: python
+
+    files = os.listdir("/content/esRedditJson/")
+
+--------------------
+See the Full Example
+--------------------
+
+.. code:: python
+    :emphasize-lines: 10,15
+
+    import os
+
+    import lightning as L
```

Comment:
This can be read from the example files directly. No need for code duplication.

Reply:
This was included at the suggestion of @edenlightning

Reply:
what example files? @tchaton

Reply:
Agree we don't need this